test/unit/reductions/moments/test_moments_bounded_group_loss.py

```diff
@@ -0,0 +1,94 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import numpy as np
+import pandas as pd
+from sklearn.linear_model import LinearRegression
+from sklearn.model_selection import train_test_split
+
+from fairlearn.reductions import GroupLossMoment, SquareLoss
+from test.unit.reductions.grid_search.utilities import _quick_data
+
+
+class TestBoundedGroupLoss:
+
+    def test_eps(self):
+        X, Y, A = _quick_data()
+        eps = 0.01
+        self.estimator = LinearRegression().fit(X, Y)
+        def predictor(x): return self.estimator.predict(x)
+        self.disparity_criterion = GroupLossMoment(SquareLoss(-np.inf, np.inf), upper_bound=eps)
+        self.disparity_criterion.load_data(X, Y, sensitive_features=A)
+        bnd = self.disparity_criterion.bound()
+        loss_eps = self.disparity_criterion.gamma(predictor) - bnd
+        loss = self.disparity_criterion.gamma(predictor)
+        assert(np.all(np.isclose(loss - eps, loss_eps)))
+
+    def test_bgl_gpa_data(self):
+
+        names = ['gender', 'physics', 'biology', 'history', 'second_language',
+                 'geography', 'literature', 'portuguese', 'math', 'chemistry', 'gpa']
+        data = pd.DataFrame(np.array([[0, 622.6, 491.56, 439.93, 707.64, 663.65, 557.09, 711.37,
```

Comment:
is this made-up data? i just want to make sure that we don't copy-paste data from external data sets unless the data sets are public domain. also, please fix the formatting: the first line break after `np.array([` and then one row of the matrix per line.

Reply:
No, it's not made-up data. It's a small dataframe generated from the dataset (https://dataverse.harvard.edu/dataset.xhtml?persistentId=doi:10.7910/DVN/O35FW8#) that the paper "Preventing undesirable behavior of intelligent machines" used. Also, I had to reformat the matrix this way because I was getting flake8 errors otherwise.

Reply:
ok. no problem. i checked the data license it's CC0 (meaning public domain)--i also see that roman offered to deal with flake8, so resolving this.